coordinator/core/core.go

```diff
@@ -162,6 +172,9 @@ func NewCore(dnsNames []string, qv quote.Validator, qi quote.Issuer, sealer seal
 		if err := txdata.putState(stateAcceptingManifest); err != nil {
 			return nil, err
 		}
+		if c.metrics != nil {
```

Comment:
We should get rid of all these nil checks and use the null object pattern instead. To this end, you could implement the prometheus.Gauge interface with a set of noop methods and use this in newCoreMetrics if factory == nil.

Reply:
Since I've changed the `coordinatorState` metric to use `GaugeFunc` instead, I've only added the null object pattern for the `marbleAPI` metrics.